### PR TITLE
ContextMenu / Menu: Apply New Data set in Context Menu and Menu component in all JS Demos

### PR DESCRIPTION
**Pull request by nikkithelegendarypokemonster**

### CODE REVIEW

File: apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts [hunk 1]
```diff
@@ -2,1243 +2,346 @@ import { Injectable } from '@angular/core';
 
 export class Product {
   text: string;
-
+
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts]: Let's remove white spaces here

File: apps/demos/Demos/ContextMenu/Scrolling/Angular/app/app.service.ts [hunk 2]
```diff
@@ -2,1243 +2,344 @@ import { Injectable } from '@angular/core';
 
 export class ContextMenuItems {
   text: string;
-
-  items: ContextMenuItems[];
+
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/ContextMenu/Scrolling/Angular/app/app.service.ts]: Let's revert this change too

Comment (nikkithelegendarypokemonster) [reply apps/demos/Demos/ContextMenu/Scrolling/Angular/app/app.service.ts]: [c076b11](https://github.com/DevExpress/DevExtreme/pull/27367/commits/c076b116b3302bdc6cb4017afe1948f85476031b)

File: apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts (hunk 1 above)
Comment (nikkithelegendarypokemonster) [reply apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts]: [c076b11](https://github.com/DevExpress/DevExtreme/pull/27367/commits/c076b116b3302bdc6cb4017afe1948f85476031b)

File: apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts [hunk 3]
```diff
@@ -2,1243 +2,345 @@ import { Injectable } from '@angular/core';
 
 export class Product {
   text: string;
-
   items?: Product[];
 }
```
Comment (EugeniyKiyashko) [inline apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts:7]: I don't see the failed tests for contextMenu demo. Could you please add tests for the demo, where contextMenu component is open?

Comment (nikkithelegendarypokemonster) [reply apps/demos/Demos/Menu/Scrolling/Angular/app/app.service.ts:7]: [3d154d2](https://github.com/DevExpress/DevExtreme/pull/27367/commits/3d154d27d4e44945263a101fa28db63207db580c)

[2b314e7](https://github.com/DevExpress/DevExtreme/pull/27367/commits/2b314e7a578bad32a37a410034304182d9994fd9)

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by iBat — APPROVED** [review]
